src/App.jsx

```diff
@@ -88,6 +92,31 @@ function App() {
       });
   }, []);
 
+  const handleMouseScroll = (event) => {
+    if(window.innerHeight + event.target.documentElement.scrollTop + 1 >= event.target.documentElement.scrollHeight){
+        setLoadMorePosts(true)
+    }
+  }
+
+useEffect(()=>{
+  if(loadMorePosts && posts.length){
+    db.collection("posts")
+      .orderBy("timestamp", "desc")
+      .startAfter(posts[posts.length-1].post.timestamp)
+      .limit(pageSize)
+      .onSnapshot((snapshot) => {
+        setPosts([
```

Comment:
i dont really understand what do you mean by saying we are not using the latest posts. since posts are sorted by timestamp when we fetch new ones we fetch the ones after this timestamp. when the user reaches scroll end again it fetches the next ones after the timestamp of the last post of the posts we fetched previously. when we reach the end there are simply no more posts left to fetch. I might be understanding something wrong though could you please explain it a little more?

Reply:
I was actually referring to the way you were updating the state. Check out the code sample, I commented.
File: App.jsx, line: 108-114.

Reply:
Yeah you seem right @mathiasayivor